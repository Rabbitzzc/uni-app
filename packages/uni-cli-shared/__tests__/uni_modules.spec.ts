import path from 'path'
import {
  findEncryptUniModules,
  findUploadEncryptUniModulesFiles,
} from '../src/uni_modules'

const platforms = ['app-android', 'app-ios', 'web'] as const
describe('uni_modules:uni-ext-api', () => {
  const inputDir = path.resolve(__dirname, '../../playground/uni_modules/src')
  test('findEncryptUniModules', () => {
    expect(findEncryptUniModules(inputDir)).toStrictEqual({
      'test-com1': undefined,
      'test-com2': undefined,
    })
  })

  platforms.forEach((platform) => {
    test(`findUploadEncryptUniModulesFiles(${platform})`, () => {
      const modules = findUploadEncryptUniModulesFiles(
        findEncryptUniModules(inputDir),
        platform,
        inputDir
      )
      expect(
        Object.keys(modules).reduce((res: string[], id: string) => {
          res.push(
            ...modules[id].map((item) => item.replace(inputDir, '').slice(1))
          )
          return res
        }, [])
      ).toMatchSnapshot()
    })
  })
})
