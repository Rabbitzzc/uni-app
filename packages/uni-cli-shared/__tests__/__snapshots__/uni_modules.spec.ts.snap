// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`uni_modules:uni-ext-api findUploadEncryptUniModulesFiles(app-android) 1`] = `
[
  "uni_modules/test-com1/encrypt",
  "uni_modules/test-com1/index.module.uts",
  "uni_modules/test-com1/package.json",
  "uni_modules/test-com1/components/test-com1-1/logo.png",
  "uni_modules/test-com1/components/test-com1-1/test-com1-1.vue",
  "uni_modules/test-com1/components/test-com1-2/test-com1-2.vue",
  "uni_modules/test-com2/encrypt",
  "uni_modules/test-com2/index.module.uts",
  "uni_modules/test-com2/index.uts",
  "uni_modules/test-com2/package.json",
  "uni_modules/test-com2/utils/index.uts",
  "uni_modules/test-com2/components/test-com2-1/api.png",
  "uni_modules/test-com2/components/test-com2-1/test-com2-1.vue",
  "uni_modules/test-com2/components/test-com2-2/test-com2-2.vue",
]
`;

exports[`uni_modules:uni-ext-api findUploadEncryptUniModulesFiles(app-ios) 1`] = `
[
  "uni_modules/test-com1/encrypt",
  "uni_modules/test-com1/index.module.uts",
  "uni_modules/test-com1/package.json",
  "uni_modules/test-com1/components/test-com1-1/test-com1-1.vue",
  "uni_modules/test-com1/components/test-com1-2/test-com1-2.vue",
  "uni_modules/test-com2/encrypt",
  "uni_modules/test-com2/index.module.uts",
  "uni_modules/test-com2/index.uts",
  "uni_modules/test-com2/package.json",
  "uni_modules/test-com2/utils/index.uts",
  "uni_modules/test-com2/components/test-com2-1/test-com2-1.vue",
  "uni_modules/test-com2/components/test-com2-2/test-com2-2.vue",
]
`;

exports[`uni_modules:uni-ext-api findUploadEncryptUniModulesFiles(web) 1`] = `
[
  "uni_modules/test-com1/encrypt",
  "uni_modules/test-com1/index.module.uts",
  "uni_modules/test-com1/package.json",
  "uni_modules/test-com1/components/test-com1-1/test-com1-1.vue",
  "uni_modules/test-com1/components/test-com1-2/test-com1-2.vue",
  "uni_modules/test-com2/encrypt",
  "uni_modules/test-com2/index.module.uts",
  "uni_modules/test-com2/index.uts",
  "uni_modules/test-com2/package.json",
  "uni_modules/test-com2/utils/index.uts",
  "uni_modules/test-com2/components/test-com2-1/test-com2-1.vue",
  "uni_modules/test-com2/components/test-com2-2/test-com2-2.vue",
]
`;
